build.sc

```diff
@@ -416,6 +422,38 @@ trait ScalaParse extends SbtModule with ScalaCliPublishModule with settings.Scal
   def scalaVersion = Scala.defaultInternal
 }
 
+trait Scala3Runtime extends SbtModule with ScalaCliPublishModule with settings.ScalaCliCompile {
+  def ivyDeps      = super.ivyDeps() ++ Agg(Deps.scalaparse)
+  def scalaVersion = Scala.scala3
+  def jar = T {
+    val original = super.jar().path
+    // scala3RuntimeFixes.jar is also used within
+    // resource-config.json and BytecodeProcessor.scala
+    val dest = original / os.up / "scala3RuntimeFixes.jar"
+    os.copy(original, dest)
+    PathRef(dest)
+  }
+}
+
+class Scala3Graal(val crossScalaVersion: String) extends BuildLikeModule {
+  def ivyDeps = super.ivyDeps() ++ Agg(
+    Deps.asm,
+    Deps.osLib
+  )
+
+  def resources = T.sources {
+    super.resources() ++ Seq(`scala3-runtime`.jar())
+  }
+}
+
+trait Scala3GraalProcessor extends ScalaModule {
+  def moduleDeps   = Seq(`scala3-graal`(Scala.scala3))
+  def scalaVersion = Scala.scala3
+  // Make sure that we put jar on run classpath
+  def runClasspath   = T(`scala3-graal`(Scala.scala3).jar() +: super.runClasspath())
```

Comment:
Why do we need this? Why isn't the `moduleDeps` enough?

Reply:
there was a porblem with hadling resources, namely jar was not visible. I guess this may be caused by my setup of resources, so I will roll it back and see if it works